### ISSUE TITLE
Problem with installation

### ISSUE THREAD

**Issue by srolskyi** [dimitern/xmlstarlet#118]
Hi there,
python version 3.8.5

run command: `pip3 install xmlstarlet`

Got an error:

```
Collecting xmlstarlet
  Using cached xmlstarlet-1.6.3.tar.gz (657 kB)
Requirement already satisfied, skipping upgrade: cffi>=1.0.0 in /opt/python38/lib/python3.8/site-packages (from xmlstarlet) (1.14.2)
Requirement already satisfied, skipping upgrade: pycparser in /opt/python38/lib/python3.8/site-packages (from cffi>=1.0.0->xmlstarlet) (2.20)
Building wheels for collected packages: xmlstarlet
  Building wheel for xmlstarlet (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /opt/python38/bin/python3.8 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-jwjvvpi6/xmlstarlet/setup.py'"'"'; __file__='"'"'/tmp/pip-install-jwjvvpi6/xmlstarlet/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-w0i7_0_6
       cwd: /tmp/pip-install-jwjvvpi6/xmlstarlet/
  Complete output (52 lines):
  + cd xmlstarlet/
  + test -f config.h
  + make -j
  Makefile:878: src/.deps/c14n-usage.Po: No such file or directory
  Makefile:879: src/.deps/depyx-usage.Po: No such file or directory
  Makefile:880: src/.deps/edit-usage.Po: No such file or directory
  Makefile:881: src/.deps/elem-usage.Po: No such file or directory
  Makefile:882: src/.deps/escape-usage.Po: No such file or directory
  Makefile:883: src/.deps/format-usage.Po: No such file or directory
  Makefile:884: src/.deps/ls-usage.Po: No such file or directory
  Makefile:885: src/.deps/pyx-usage.Po: No such file or directory
  Makefile:886: src/.deps/select-usage.Po: No such file or directory
  Makefile:887: src/.deps/trans-usage.Po: No such file or directory
  Makefile:888: src/.deps/trans.Po: No such file or directory
  Makefile:889: src/.deps/unescape-usage.Po: No such file or directory
  Makefile:890: src/.deps/usage.Po: No such file or directory
  Makefile:891: src/.deps/validate-usage.Po: No such file or directory
  Makefile:892: src/.deps/xml.Po: No such file or directory
  Makefile:893: src/.deps/xml_C14N.Po: No such file or directory
  Makefile:894: src/.deps/xml_depyx.Po: No such file or directory
  Makefile:895: src/.deps/xml_edit.Po: No such file or directory
  Makefile:896: src/.deps/xml_elem.Po: No such file or directory
  Makefile:897: src/.deps/xml_escape.Po: No such file or directory
  Makefile:898: src/.deps/xml_format.Po: No such file or directory
  Makefile:899: src/.deps/xml_ls.Po: No such file or directory
  Makefile:900: src/.deps/xml_pyx.Po: No such file or directory
  Makefile:901: src/.deps/xml_select.Po: No such file or directory
  Makefile:902: src/.deps/xml_trans.Po: No such file or directory
  Makefile:903: src/.deps/xml_validate.Po: No such file or directory
  make: *** No rule to make target `src/.deps/xml_validate.Po'.  Stop.
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.8
  creating build/lib.linux-x86_64-3.8/xmlstarlet
  copying xmlstarlet/__init__.py -> build/lib.linux-x86_64-3.8/xmlstarlet
  copying xmlstarlet/xmlstarlet_build.py -> build/lib.linux-x86_64-3.8/xmlstarlet
  running build_ext
  generating cffi module 'build/temp.linux-x86_64-3.8/xmlstarlet._xmlstarlet.c'
  creating build/temp.linux-x86_64-3.8
  building 'xmlstarlet._xmlstarlet' extension
  creating build/temp.linux-x86_64-3.8/build
  creating build/temp.linux-x86_64-3.8/build/temp.linux-x86_64-3.8
  creating build/temp.linux-x86_64-3.8/xmlstarlet
  creating build/temp.linux-x86_64-3.8/xmlstarlet/src
  gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/usr/include -I/usr/include/libxml2 -I/tmp/pip-install-jwjvvpi6/xmlstarlet/xmlstarlet -I/tmp/pip-install-jwjvvpi6/xmlstarlet/xmlstarlet/src -I/opt/python38/include/python3.8 -c build/temp.linux-x86_64-3.8/xmlstarlet._xmlstarlet.c -o build/temp.linux-x86_64-3.8/build/temp.linux-x86_64-3.8/xmlstarlet._xmlstarlet.o
  build/temp.linux-x86_64-3.8/xmlstarlet._xmlstarlet.c:570:10: fatal error: libxml/xmlversion.h: No such file or directory
   #include <libxml/xmlversion.h>
            ^~~~~~~~~~~~~~~~~~~~~
  compilation terminated.
  error: command 'gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for xmlstarlet
  Running setup.py clean for xmlstarlet
Failed to build xmlstarlet
Installing collected packages: xmlstarlet
    Running setup.py install for xmlstarlet ... error
    ERROR: Command errored out with exit status 1:
     command: /opt/python38/bin/python3.8 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-jwjvvpi6/xmlstarlet/setup.py'"'"'; __file__='"'"'/tmp/pip-install-jwjvvpi6/xmlstarlet/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-f56watal/install-record.txt --single-version-externally-managed --compile --install-headers /opt/python38/include/python3.8/xmlstarlet
         cwd: /tmp/pip-install-jwjvvpi6/xmlstarlet/
    Complete output (52 lines):
    + cd xmlstarlet/
    + test -f config.h
    + make -j
    Makefile:878: src/.deps/c14n-usage.Po: No such file or directory
    Makefile:879: src/.deps/depyx-usage.Po: No such file or directory
    Makefile:880: src/.deps/edit-usage.Po: No such file or directory
    Makefile:881: src/.deps/elem-usage.Po: No such file or directory
    Makefile:882: src/.deps/escape-usage.Po: No such file or directory
    Makefile:883: src/.deps/format-usage.Po: No such file or directory
    Makefile:884: src/.deps/ls-usage.Po: No such file or directory
    Makefile:885: src/.deps/pyx-usage.Po: No such file or directory
    Makefile:886: src/.deps/select-usage.Po: No such file or directory
    Makefile:887: src/.deps/trans-usage.Po: No such file or directory
    Makefile:888: src/.deps/trans.Po: No such file or directory
    Makefile:889: src/.deps/unescape-usage.Po: No such file or directory
    Makefile:890: src/.deps/usage.Po: No such file or directory
    Makefile:891: src/.deps/validate-usage.Po: No such file or directory
    Makefile:892: src/.deps/xml.Po: No such file or directory
    Makefile:893: src/.deps/xml_C14N.Po: No such file or directory
    Makefile:894: src/.deps/xml_depyx.Po: No such file or directory
    Makefile:895: src/.deps/xml_edit.Po: No such file or directory
    Makefile:896: src/.deps/xml_elem.Po: No such file or directory
    Makefile:897: src/.deps/xml_escape.Po: No such file or directory
    Makefile:898: src/.deps/xml_format.Po: No such file or directory
    Makefile:899: src/.deps/xml_ls.Po: No such file or directory
    Makefile:900: src/.deps/xml_pyx.Po: No such file or directory
    Makefile:901: src/.deps/xml_select.Po: No such file or directory
    Makefile:902: src/.deps/xml_trans.Po: No such file or directory
    Makefile:903: src/.deps/xml_validate.Po: No such file or directory
    make: *** No rule to make target `src/.deps/xml_validate.Po'.  Stop.
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.8
    creating build/lib.linux-x86_64-3.8/xmlstarlet
    copying xmlstarlet/__init__.py -> build/lib.linux-x86_64-3.8/xmlstarlet
    copying xmlstarlet/xmlstarlet_build.py -> build/lib.linux-x86_64-3.8/xmlstarlet
    running build_ext
    generating cffi module 'build/temp.linux-x86_64-3.8/xmlstarlet._xmlstarlet.c'
    creating build/temp.linux-x86_64-3.8
    building 'xmlstarlet._xmlstarlet' extension
    creating build/temp.linux-x86_64-3.8/build
    creating build/temp.linux-x86_64-3.8/build/temp.linux-x86_64-3.8
    creating build/temp.linux-x86_64-3.8/xmlstarlet
    creating build/temp.linux-x86_64-3.8/xmlstarlet/src
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/usr/include -I/usr/include/libxml2 -I/tmp/pip-install-jwjvvpi6/xmlstarlet/xmlstarlet -I/tmp/pip-install-jwjvvpi6/xmlstarlet/xmlstarlet/src -I/opt/python38/include/python3.8 -c build/temp.linux-x86_64-3.8/xmlstarlet._xmlstarlet.c -o build/temp.linux-x86_64-3.8/build/temp.linux-x86_64-3.8/xmlstarlet._xmlstarlet.o
    build/temp.linux-x86_64-3.8/xmlstarlet._xmlstarlet.c:570:10: fatal error: libxml/xmlversion.h: No such file or directory
     #include <libxml/xmlversion.h>
              ^~~~~~~~~~~~~~~~~~~~~
    compilation terminated.
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /opt/python38/bin/python3.8 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-jwjvvpi6/xmlstarlet/setup.py'"'"'; __file__='"'"'/tmp/pip-install-jwjvvpi6/xmlstarlet/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-f56watal/install-record.txt --single-version-externally-managed --compile --install-headers /opt/python38/include/python3.8/xmlstarlet Check the logs for full command output.
```

**Comment by dimitern** [comment]
Hey @srolskyi ! Thanks for reporting the issue, I'll have a look and try to come up with a solution for Python 3.8.5 and let you know!

**Comment by srolskyi** [comment]
@dimitern thank you!

btw, I tried to install this tools for another VM and another python version

Python 3.6.8
run command: `pip3 install xmlstarlet`

And I got an error too...

```
Collecting xmlstarlet
  Using cached https://files.pythonhosted.org/packages/a9/58/e7bf49516b285130cf38689f47afd6446642ad2f82b885397487b2e3c386/xmlstarlet-1.6.3.tar.gz
Requirement already satisfied: cffi>=1.0.0 in /usr/local/lib64/python3.6/site-packages (from xmlstarlet)
Requirement already satisfied: pycparser in /usr/local/lib/python3.6/site-packages (from cffi>=1.0.0->xmlstarlet)
Building wheels for collected packages: xmlstarlet
  Running setup.py bdist_wheel for xmlstarlet ... error
  Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-xvmusc37/xmlstarlet/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmp3zviri8_pip-wheel- --python-tag cp36:
  + cd xmlstarlet/
  + test -f config.h
  + make -j
  Makefile:878: src/.deps/c14n-usage.Po: No such file or directory
  Makefile:879: src/.deps/depyx-usage.Po: No such file or directory
  Makefile:880: src/.deps/edit-usage.Po: No such file or directory
  Makefile:881: src/.deps/elem-usage.Po: No such file or directory
  Makefile:882: src/.deps/escape-usage.Po: No such file or directory
  Makefile:883: src/.deps/format-usage.Po: No such file or directory
  Makefile:884: src/.deps/ls-usage.Po: No such file or directory
  Makefile:885: src/.deps/pyx-usage.Po: No such file or directory
  Makefile:886: src/.deps/select-usage.Po: No such file or directory
  Makefile:887: src/.deps/trans-usage.Po: No such file or directory
  Makefile:888: src/.deps/trans.Po: No such file or directory
  Makefile:889: src/.deps/unescape-usage.Po: No such file or directory
  Makefile:890: src/.deps/usage.Po: No such file or directory
  Makefile:891: src/.deps/validate-usage.Po: No such file or directory
  Makefile:892: src/.deps/xml.Po: No such file or directory
  Makefile:893: src/.deps/xml_C14N.Po: No such file or directory
  Makefile:894: src/.deps/xml_depyx.Po: No such file or directory
  Makefile:895: src/.deps/xml_edit.Po: No such file or directory
  Makefile:896: src/.deps/xml_elem.Po: No such file or directory
  Makefile:897: src/.deps/xml_escape.Po: No such file or directory
  Makefile:898: src/.deps/xml_format.Po: No such file or directory
  Makefile:899: src/.deps/xml_ls.Po: No such file or directory
  Makefile:900: src/.deps/xml_pyx.Po: No such file or directory
  Makefile:901: src/.deps/xml_select.Po: No such file or directory
  Makefile:902: src/.deps/xml_trans.Po: No such file or directory
  Makefile:903: src/.deps/xml_validate.Po: No such file or directory
  make: *** No rule to make target `src/.deps/xml_validate.Po'.  Stop.
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.6
  creating build/lib.linux-x86_64-3.6/xmlstarlet
  copying xmlstarlet/__init__.py -> build/lib.linux-x86_64-3.6/xmlstarlet
  copying xmlstarlet/xmlstarlet_build.py -> build/lib.linux-x86_64-3.6/xmlstarlet
  running build_ext
  generating cffi module 'build/temp.linux-x86_64-3.6/xmlstarlet._xmlstarlet.c'
  creating build/temp.linux-x86_64-3.6
  building 'xmlstarlet._xmlstarlet' extension
  creating build/temp.linux-x86_64-3.6/build
  creating build/temp.linux-x86_64-3.6/build/temp.linux-x86_64-3.6
  creating build/temp.linux-x86_64-3.6/xmlstarlet
  creating build/temp.linux-x86_64-3.6/xmlstarlet/src
  gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/include -I/usr/include/libxml2 -I/tmp/pip-build-xvmusc37/xmlstarlet/xmlstarlet -I/tmp/pip-build-xvmusc37/xmlstarlet/xmlstarlet/src -I/usr/include/python3.6m -c build/temp.linux-x86_64-3.6/xmlstarlet._xmlstarlet.c -o build/temp.linux-x86_64-3.6/build/temp.linux-x86_64-3.6/xmlstarlet._xmlstarlet.o
  build/temp.linux-x86_64-3.6/xmlstarlet._xmlstarlet.c:493:31: fatal error: libxml/xmlversion.h: No such file or directory
   #include <libxml/xmlversion.h>
                                 ^
  compilation terminated.
  error: command 'gcc' failed with exit status 1
  
  ----------------------------------------
  Failed building wheel for xmlstarlet
  Running setup.py clean for xmlstarlet
Failed to build xmlstarlet
Installing collected packages: xmlstarlet
  Running setup.py install for xmlstarlet ... error
    Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-xvmusc37/xmlstarlet/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-4l00db11-record/install-record.txt --single-version-externally-managed --compile:
    + cd xmlstarlet/
    + test -f config.h
    + make -j
    Makefile:878: src/.deps/c14n-usage.Po: No such file or directory
    Makefile:879: src/.deps/depyx-usage.Po: No such file or directory
    Makefile:880: src/.deps/edit-usage.Po: No such file or directory
    Makefile:881: src/.deps/elem-usage.Po: No such file or directory
    Makefile:882: src/.deps/escape-usage.Po: No such file or directory
    Makefile:883: src/.deps/format-usage.Po: No such file or directory
    Makefile:884: src/.deps/ls-usage.Po: No such file or directory
    Makefile:885: src/.deps/pyx-usage.Po: No such file or directory
    Makefile:886: src/.deps/select-usage.Po: No such file or directory
    Makefile:887: src/.deps/trans-usage.Po: No such file or directory
    Makefile:888: src/.deps/trans.Po: No such file or directory
    Makefile:889: src/.deps/unescape-usage.Po: No such file or directory
    Makefile:890: src/.deps/usage.Po: No such file or directory
    Makefile:891: src/.deps/validate-usage.Po: No such file or directory
    Makefile:892: src/.deps/xml.Po: No such file or directory
    Makefile:893: src/.deps/xml_C14N.Po: No such file or directory
    Makefile:894: src/.deps/xml_depyx.Po: No such file or directory
    Makefile:895: src/.deps/xml_edit.Po: No such file or directory
    Makefile:896: src/.deps/xml_elem.Po: No such file or directory
    Makefile:897: src/.deps/xml_escape.Po: No such file or directory
    Makefile:898: src/.deps/xml_format.Po: No such file or directory
    Makefile:899: src/.deps/xml_ls.Po: No such file or directory
    Makefile:900: src/.deps/xml_pyx.Po: No such file or directory
    Makefile:901: src/.deps/xml_select.Po: No such file or directory
    Makefile:902: src/.deps/xml_trans.Po: No such file or directory
    Makefile:903: src/.deps/xml_validate.Po: No such file or directory
    make: *** No rule to make target `src/.deps/xml_validate.Po'.  Stop.
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.6
    creating build/lib.linux-x86_64-3.6/xmlstarlet
    copying xmlstarlet/__init__.py -> build/lib.linux-x86_64-3.6/xmlstarlet
    copying xmlstarlet/xmlstarlet_build.py -> build/lib.linux-x86_64-3.6/xmlstarlet
    running build_ext
    generating cffi module 'build/temp.linux-x86_64-3.6/xmlstarlet._xmlstarlet.c'
    creating build/temp.linux-x86_64-3.6
    building 'xmlstarlet._xmlstarlet' extension
    creating build/temp.linux-x86_64-3.6/build
    creating build/temp.linux-x86_64-3.6/build/temp.linux-x86_64-3.6
    creating build/temp.linux-x86_64-3.6/xmlstarlet
    creating build/temp.linux-x86_64-3.6/xmlstarlet/src
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/include -I/usr/include/libxml2 -I/tmp/pip-build-xvmusc37/xmlstarlet/xmlstarlet -I/tmp/pip-build-xvmusc37/xmlstarlet/xmlstarlet/src -I/usr/include/python3.6m -c build/temp.linux-x86_64-3.6/xmlstarlet._xmlstarlet.c -o build/temp.linux-x86_64-3.6/build/temp.linux-x86_64-3.6/xmlstarlet._xmlstarlet.o
    build/temp.linux-x86_64-3.6/xmlstarlet._xmlstarlet.c:493:31: fatal error: libxml/xmlversion.h: No such file or directory
     #include <libxml/xmlversion.h>
                                   ^
    compilation terminated.
    error: command 'gcc' failed with exit status 1
    
    ----------------------------------------
Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-xvmusc37/xmlstarlet/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-4l00db11-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-xvmusc37/xmlstarlet/
```

**Comment by srolskyi** [comment]
Hello @dimitern, any ideas?

**Comment by dimitern** [comment]
Hey @srolskyi !

Sorry, I only had the time  to look into it today.
This is what I did to get it to build, but not from the pypi release, but from a source checkout.

```bash
$ git clone https://github.com/dimitern/xmlstarlet.git
$ pyenv install 3.8.5  # optional
$ pyenv local 3.8.5   # optional
# The following is needed to be able to build the C sources of xmlstarlet:
$ sudo apt install libxml2-dev libxslt1-dev   # for Ubuntu
$ python -m venv  .venv
$ source .venv/bin/activate
# The following 2 steps were needed to solve 
$ cat requirements.txt | sed -e 's/wheel==0.31.1/wheel==0.34.2/g' > requirements-fixed.txt
$ pip install -r requirements-fixed.txt
$ python setup.py install
$ python setup.py test   # optional, to verify the build
```

I need to make a new release and try to see if it will work better at least for some platforms with binary wheels.
Until then, I hope this helps!

**Comment by srolskyi** [comment]
That helped me, thanks.

**Comment by dimitern** [comment]
Hey @srolskyi ! I've just released v1.6.5 to pypi, can you please test if it works for you without any workarounds like above? Thanks!

**Comment by dimitern** [comment]
Fixed with #131, once 1.6.6 is released this will be closed. Please, try after the release and reopen if you still have issues.
